File: common/models/kanji.go
// Package models contains the models shared amongst multiple go services
package models

type Kanji struct {
	Character string `bson:"character" json:"character"`
	WaniKaniId int `bson:"wanikani_id" json:"wanikani_id"`
	WaniKanilevel int `bson:"wanikani_level" json:"wanikani_level"`
	Meanings []string `bson:"meanings" json:"meanings"`
	Onyomi []string `bson:"onyomi" json:"onyomi"`
	Kunyomi []string `bson:"kunyomi" json:"kunyomi"`
	Nanori []string `bson:"nanori" json:"nanori"`
}

File: services/web/src/utils/graphql/schema/kanji.ts
import { gql } from "@apollo/client";
import { DocumentNode } from "graphql";

function queryGetKanjiByLevelRangeDocument(lowerBound: number, upperBound: number): DocumentNode {
	let mutationString = `
	query {
    getKanjiByLevelRange(lowerBound: ${lowerBound}, upperBound: ${upperBound}) {
      Character,
      WanikaniId,
      WanikaniLevel,
      Meanings,
      Onyomi,
      Kunyomi,
      Nanori
    }
  }`;
	return gql(mutationString);
}

export const kanji = {
	queryGetKanjiByLevelRangeDocument,
}